fetch/client/src/test/scala/endpoints4s/fetch/future/FetchClientEndpointsJsonSchemaTest.scala

```diff
@@ -0,0 +1,44 @@
+package endpoints4s.fetch.future
+
+import endpoints4s.algebra.JsonTestApi
+import endpoints4s.algebra.client.JsonTestSuite
+import endpoints4s.algebra.Address
+import endpoints4s.algebra.User
+import endpoints4s.fetch.BasicAuthentication
+import endpoints4s.fetch.EndpointsSettings
+import endpoints4s.fetch.JsonEntitiesFromSchemas
+import endpoints4s.generic
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.ExecutionContextExecutor
+import scala.concurrent.Future
+import scala.scalajs.concurrent.JSExecutionContext
+
+class TestJsonSchemaClient(val settings: EndpointsSettings)(implicit
+    val ec: ExecutionContext
+) extends Endpoints
+    with BasicAuthentication
+    with BasicAuthenticationTestApi
+    with generic.JsonSchemas
+    with JsonTestApi
+    with JsonEntitiesFromSchemas {
+  implicit def userCodec: JsonSchema[User] = genericJsonSchema[User]
+  implicit def addresCodec: JsonSchema[Address] = genericJsonSchema[Address]
+}
+
+class FetchClientEndpointsJsonSchemaTest
+    extends JsonTestSuite[TestJsonSchemaClient]
+    with BasicAuthTestSuite[TestJsonSchemaClient] {
+
+  implicit override def executionContext: ExecutionContextExecutor = JSExecutionContext.queue
+
+  val client: TestJsonSchemaClient = new TestJsonSchemaClient(
+    EndpointsSettings()
+      .withHost(Some("http://localhost:8080"))
+  )
+
+  def call[Req, Resp](
+      endpoint: client.Endpoint[Req, Resp],
+      args: Req
+  ): Future[Resp] = endpoint(args)
+}
```

Comment:
Are you sure some tests are executed here?

Reply:
Sorry to be nitpicking again here, but it seems that `basicAuthSuite()` is redundant with what we have in `FetchClientEndpointsTest`, no?

Reply:
It is alright, I am just copypasting things from XHR and Akka HTTP clients, got to remember to fix these same issues there too.